PYME/Acquire/ui/intensity_trace.py

```diff
@@ -0,0 +1,111 @@
+
+import wx
+import numpy as np
+from PYME.ui.fastGraph import FastGraphPanel
+import weakref
+from PYME.ui.selection import SELECTION_RECTANGLE
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class IntensityTracePanel(FastGraphPanel):
+    def __init__(self, parent, frame_wrangler, winid=-1, n_frames=1000):
+        """If the square region select tool is active, will plot the average value within it
+        over time. Note that the update rate is set by the GUI timer, not the camera frame rate.
+
+        Parameters
+        ----------
+        parent : wx.Window
+            PYMEAcquire MainFrame / AUIFrame
+        frame_wrangler : PYME.Acquire.FrameWrangler
+            frame_wrangler object, which is grabbing new frames from the camera
+        winid : int, optional
+            direct pass-through to FastGraphPanel, by default -1
+        n_frames : int, optional
+            Number of frame-updates to store and display, by default 1000
+        """
+        self.frame_vals = np.arange(n_frames)
+        self.intensity_avg = np.zeros_like(self.frame_vals, dtype=float)
+        FastGraphPanel.__init__(self, parent, winid, self.frame_vals, 
+                                self.intensity_avg)
+        self.wrangler = frame_wrangler
+        # differ display opts assignment because it might not have been created yet
+        self.do = None  # PYME.UI.displayOps.DisplayOptions gets assigned later
+        self._mf = weakref.ref(parent)  # weak ref to MainFrame
+        self._relative_val = 1
+
+    def assign_do(self):

```

Comment:
this is cheap - what about making it a property instead of the testing and assign if not present logic (which is missing from `clear()`)